### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by Maetveis**
I have used this template for my thesis, and made some improvements to the template during that.

This is not yet in a clean state, I just wanted a review, if this is a good direction in general.

Summary of changes:
- Replaced obsolete packages with equivalents.
- Fixed issues with too large spacing in tables, in vectors and matrices
- ~~Move the table of content generation of listings to the class file~~, does not work correctly reverted
- Move the contents of settings.tex to the class file
- Enable correct compilation in UTF8 capable Tex Engines (LuaTex and XeTex), tested on LuaTex
- Use the minted package for source code highlighting, this requires a pip module to be installed
   - TODO: fix the CI, add installation to the documentation
   - TODO: make optional?

### CODE REVIEW

File: elteikthesis.cls [hunk 1]
```diff
@@ -247,7 +256,6 @@
 
 % ------------------------------
 % Helyközök és tördelés
-\linespread{1.5}
```
Comment (mcserep) [inline elteikthesis.cls:250]: Is there an additional benefit of setting the line spread through the `setspace` package? If not, the simpler version should be kept.

File: elteikthesis.cls [hunk 2]
```diff
@@ -256,10 +264,11 @@
 \nottoggle{hypenation}{\RequirePackage[none]{hyphenat}}{}
 
 % Beállítás alternatív módon
-%\RequirePackage{setspace}
-%\singlespacing		% ~1.0 linespread
-%\onehalfspacing	% ~1.3 linespread
-%\doublespacing		% ~1.6 linespread
+\RequirePackage{setspace}
+\setstretch{1.5}
```
Comment (mcserep) [inline elteikthesis.cls:268]: The line spacing should be 1.3, as that is somewhat equivalent to 1.5 line spacing in MS Word. It seems it was not correct   so far.

**Review by mcserep — CHANGES_REQUESTED** [review]
Thanks for your contribution in advance. I generally agree with your changes. Replacing obsolete packages and testing on LuaTex (as I have used pdfTeX mostly) are especially useful for the template. The only part of the changes which I disagree with are the modified spacing in tables. Now there is only a very small padding, the borders almost touch the text, which looks like a typographical error to me. The spacing in tables should be similar to the line spacing of the whole document in my opinion, otherwise the formatting of the document looks a bit inconsistent.

Code highlighting is not a strong part of the current template, so using the `minted` package is very welcomed.
I agree though that it should be optional, so Python does not become a mandatory dependency. In many cases there are no code samples in a thesis or only a few lines of configuration code is embedded. It would be nice that in such cases fallback to using *listings* would be supported.

File: elteikthesis.cls (hunk 1 above)
Comment (Maetveis) [reply elteikthesis.cls:250]: Yes, `setspace` doesn't change the spacing in 'qeustionable' like the footnotes, it also keeps single spacing in code listings, which is very ugly with 1.5 spacing imo.

**Comment by Maetveis** [comment]
The table padding is screwed up because of the workaround for vectors and matrices.
```tex
% Vektorok és mátrixok egyszeres sorközzel
\renewcommand*{\arraystretch}{0.667}
```
Without this matrices and standing vectors are displayed with 1.5 linspacing, and it looks weird.
Another solution is required for that problem, then.

File: elteikthesis.cls (hunk 2 above)
Comment (Maetveis) [reply elteikthesis.cls:268]: Acording to https://tex.stackexchange.com/questions/442030/setting-spaces-with-titlespacing-and-fonts 1.423 is the equialent of words 1.5 spacing. I checked with Libreoffice and with this setting matches the "1.5 lines" option.

Comment (mcserep) [reply elteikthesis.cls]: Somewhat funny how the definition of line spacing has so many possible interpretations. The linked reference mentions that line spacing in MS Word depends on the typeface/font and the value 1.427465 only applies for Times New Roman.

Anyway, I have checked and the formatting requirements at the faculty also do not provide any further information on the definition of line spacing, so technically both 1.5, 1.3 and 1.427465 can be valid :smile:

File: elteikthesis.cls (hunk 1 above)
Comment (mcserep) [reply elteikthesis.cls:250]: I agree that the 1.5x line spacing should not be applied on code listings and mathematical formulas.
Rules for footnotes are not really unified in this matter in my experience. E.g. the *Chicago Manual of Style* (which is one of the most well known formatting guidelines) states that text should be double-spaced, but footnotes *can* be single-spaced.

**Comment by Maetveis** [comment]
The CI update was unexpectedly trivial, the used github action already supports the minted package.
The minted package is now an option for the package (off by default), if it is disabled the listings package can be used.

I'm going to add a CI job for luatex, and maybe a test file and a job with minted disabled too.

**Comment by mcserep** [comment]
Sounds great, feel free to remove the draft status if the PR is considered complete and I shall re-review it again.

**Comment by Maetveis** [comment]
In the end I did a different integration with the minted package: the class does not load it, but if it's loaded in the document, then a default style is set, and the listing names are localized.

**Comment by Maetveis** [comment]
In the end I did a different integration with the minted package: the class does not load it, but if it's loaded in the document, then a default style is set, and the listing names are localized.

**Review by mcserep — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**